WordPress/Classes/Utility/In-App Feedback/SubmitFeedbackViewController.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+import Combine
+import DesignSystem
+
+final class SubmitFeedbackViewController: UIViewController {
+
+    // MARK: - Public Properties
+
+    private(set) var feedbackWasSubmitted = false
+
+    var onFeedbackSubmitted: ((SubmitFeedbackViewController, String) -> Void)?
+
+    // MARK: - Private Properties
+
+    private let textView: UITextView = {
+        let textView = UITextView()
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        textView.font = WPStyleGuide.fontForTextStyle(.body)
+        textView.textContainer.lineFragmentPadding = 0
+        textView.textContainerInset = .init(allEdges: Length.Padding.double)
+        return textView
+    }()
+
+    private var cancellables = Set<AnyCancellable>()
+
+    // MARK: - View Lifecycle
+
+    deinit {
+        guard !feedbackWasSubmitted else {
+            return
+        }
+        WPAnalytics.track(.appReviewsCanceledFeedbackScreen)
+    }
```

Comment:
Shouldn't we do this in `viewDidDisappear` or some other view lifecycle function rather than deinit which is about memory deallocation?

Reply:
Resolved in a9dd8ab5d8a26f8e477d71424137e8cd967b854d